go/src/koding/kites/kloud/stack/provider/builder.go

```diff
@@ -217,6 +217,20 @@ func (b *Builder) BuildStackTemplate(templateID string) error {
 	return nil
 }
 
+// Authorize verifies whether user is allowed to access b.StackTemplate.
+//
+// Prior calling to this method it is required to build the stack
+// template first.
+//
+// If the stack template was not built, the method is a nop.
+func (b *Builder) Authorize(username string) error {
+	if b.StackTemplate == nil {
```

Comment:
`Builder.StackTemplate` could be nil. It's zero value is nil. So HasTemplateAccess could get `tmpl` as nil.


> should we add checks to all functions

i dont think it is that important ATM, but we can fix places whenever we touch near them